portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java

```diff
@@ -0,0 +1,76 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/login")
+public class LoginStatusServlet extends HttpServlet {
+
+    private final UserService userService = UserServiceFactory.getUserService();
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("text/html");
+
+        String urlToRedirectToAfterUserLogsOut = "/";
+        String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+        String urlToRedirectToAfterUserLogsIn = "/index.html";
+        String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+
+        Boolean isUserLoggedIn = userService.isUserLoggedIn();
+
+        LoginData newLoginData = new LoginData(loginUrl, logoutUrl, isUserLoggedIn);
+        String json = convertToJson(newLoginData);
+
+        response.setContentType("application/json");
+        response.getWriter().println(json);
+
+    }
+
+    public static class LoginData {
+        String loginUrl;
+        String logoutUrl;
+        Boolean isUserLoggedIn;
+
+        LoginData(String loginUrl, String logoutUrl, Boolean isUserLoggedIn) {
+            this.loginUrl = loginUrl;
+            this.logoutUrl = logoutUrl;
+            this.isUserLoggedIn = isUserLoggedIn;
+        }
+    }
+
+    private String convertToJson(LoginData loginData) {
```

Comment:
why is this function necessary? doesn't the Gson class handle json conversion for you? You seem to be using that in other classes?

Reply:
I thought it only worked with array of objects. Just found out it works with objects too